### PR TITLE
Always request download URL instead of using HEAD, cleanup some tests

### PR DESCRIPTION
**Pull request by sethvargo**
It doesn't look like the components URL actually supports HEAD requests, so we end up downloading the file twice. Worse, we do it in a retry loop, so it adds a lot of time.

It's probably better just to try and download the file (in a retry).

### CODE REVIEW

File: src/version-util.ts [hunk 1]
```diff
@@ -18,35 +18,83 @@
...
+
+/**
+ * getGcloudVersion is a helper that actually downloads the gcloud version
+ * information.
+ *
+ * @param url URL for the component download. This is exposed for testing, but
+ * should be called without parameters.
+ *
+ */
+async function getGcloudVersion(url: string): Promise<string> {
```
Comment (sethvargo) [inline src/version-util.ts:56]: I had aspirations of writing unit tests for this, but spinning up an http(s) server for testing is surprisingly hard in node.

File: tests/download-util.test.ts [hunk 2]
```diff
@@ -51,11 +45,7 @@ describe('#downloadAndExtractTool', function () {
   });
 
   it('downloads and extracts linux version', async function () {
-    if (os.platform() === 'win32') {
```
Comment (sethvargo) [inline tests/download-util.test.ts:54]: This issue is from 2019, I'm curious if it's still an issue...

File: src/user-agent-util.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+// Do not listen to the linter - this can NOT be rewritten as an ES6 import statement.
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { version: appVersion } = require('../package.json');
+
+/**
+ * userAgentString is the UA to use for this installation. It dynamically pulls
+ * the app version from the package declaration.
+ */
+export const userAgentString = `google-github-actions:setup-gcloud-sdk/${appVersion}`;
```
Comment (averikitsch) [inline src/user-agent-util.ts:11]: nit
```suggestion
export const userAgentString = `google-github-actions:setup-cloud-sdk/${appVersion}`;
```

File: tests/download-util.test.ts (hunk 2 above)
Comment (averikitsch) [reply tests/download-util.test.ts:54]: I tried to remove this but it still errored. I wasn't able to investigate.

**Review by averikitsch — APPROVED** [review]

File: tests/download-util.test.ts (hunk 2 above)
Comment (sethvargo) [reply tests/download-util.test.ts]: Yea - me too. I'm going to add it back in for now to separate debugging from this small refactor.

**Pull request merged**